cpp/Forces.h

```diff
@@ -122,13 +134,121 @@ inline double addAtomLJ( const Vec3d& dR, Vec3d& fout, double c6, double c12 ){
 }
 
 // Lenard-Jones force between two atoms a,b separated by vector dR = Ra - Rb
-inline double addAtomVdW( const Vec3d& dR, Vec3d& fout, double c6 ){
+inline double addAtomVdW_noDamp( const Vec3d& dR, Vec3d& fout, double c6 ){
+    double invR2 = 1/dR.norm2(); 
+    double invR4 = invR2*invR2;
+    double E     = -c6 * invR4*invR2;
+    fout.add_mul( dR , E*-6*invR2 );
+    return E;
+}
+
+// Lenard-Jones force between two atoms a,b separated by vector dR = Ra - Rb
+inline double addAtomVdW_dampConst( const Vec3d& dR, Vec3d& fout, double c6, double ADamp ){
     double r2 = dR.norm2(); r2*=r2; r2*=r2;
-    //fout.add_mul( dR , 6*c6 /( r2 + 1.0 ) );
-    //fout.add_mul( dR , 6*c6 /( r2 + 60*c6 ) );
-    fout.add_mul( dR , 6*c6 /( r2 + 180*c6 ) );
+    fout.add_mul  ( dR , 6*c6 /( r2 + ADamp*c6 ) );
+    return 0;
+}
+
+
+
+/*
+// Lenard-Jones force between two atoms a,b separated by vector dR = Ra - Rb
+inline double addAtomVdW_dampR4( const Vec3d& dR, Vec3d& fout, double E0, double invR2cut, double cdamp ){
+    double r2 = dR.norm2(); 
+    double r8=r2*r2; r8=r8*r8;
+    double R2fun = 1-r2*invR2cut;
+    double dR2   = 1-r2*invR2cut;
+    if (damp<0) damp=0;  
+    F = Eij*r*( -C6/(r8 + cdamp*R2fun*R2fun  )  )  
+    return 0;
+}
+
+// https://manual.q-chem.com/5.2/Ch5.S7.SS2.html
+inline double addAtomVdW_Chai( const Vec3d& dR, Vec3d& fout, double E0,  double invR2cut, double cdamp ){
+    double r2 = dR.norm2()*invR2cut; 
+    double r4=r2*r2; r8=r8*r8;
+    double d  = 1/r8;
+    double D = 1./( 1. + A * D );
+    double D = 8*A*D*E*E/r**2
+
+    F = 12*A*D*E*E/r**2
+
+    double damp = 1-r2*invR2cut;  // damp = 1-(r/Rcut)^2 
+    if (damp<0) damp=0;  
+    F = Eij*r*( -C6/(r8 + cdamp/r8 )  )  
     return 0;
 }
+*/
+
+
+/*
+
+E     =     (R/r)^6
+dE_dr = r * R^6 * -6 * 1/r^8
+      = r * (R/r)^6 * -6 * 1/r^2
+      = r * (R/r)^12 * ( r^4 ) / R^6
+      =     (R/r)^12 * ( (r/R)^4 ) / (R*R)
+
+
+
+*/
+
+template<double Rfunc(double r2,double &df)>
+double addAtomVdW_addDamp( const Vec3d& dR, Vec3d& fout, double R, double E0, double ADamp ){
+    double D,dD;
+    double r2    = dR.norm2();
+    double invR2 = 1./(R*R);
+    double u2    = r2*invR2; 
+    double u4    = u2*u2; 
+    D  = Rfunc(u2,dD);
+    //ADamp=0; // DEBUG
+    double e  = 1./(      u4*u2 + D*ADamp);       
+    //printf( "r %g e %g R0 %g E0 %g \n", sqrt(r2), e,  R, E0 );             
+    double E  = -2*E0*e;
+    double fr = -E *e*( 6*u4    + dD*ADamp)*invR2 ; 
+    //fr = dD*E0; E = D*E0; // DEBUG
+    fout.add_mul(dR,fr);
```

Comment:
I'm not convinced that this formula for the force is correct. From my calculation, the exponent for `u` in the numerator should be 5 instead of 4, and to make the unit vector for `dR` I think the denominator should be `sqrt(r2)`, so something like
```C
double fr = -E *e*( 6*u4*sqrt(u2) + dD*ADamp) / sqrt(r2);
```
Also, is the sign correct? The force is the negative derivative of the energy, but you also get one minus sign from the derivative of the denominator.

Reply:
I think that would be true without the damping function, but you are also multiplying the damping function by `r` which would cause some distortion to the damping of the force.

`invR2` here is the inverse of the square of the LJ parameter, isn't it, so it's not the same as `1 / sqrt(r2) = 1 / |dR|` which I thought of as the normalization for `dR` as you said.

For the minus signs, I think there should be three, one for the energy, one for the fact that `F = -dE/dr`, and one from the derivative of `1/f(r)`. But I think you only have two, because one is already in the energy, and the other two should cancel out in the force formula.

Reply:
* Notice that  dD returned from Rfunc is also missing one power of `r` exactly for this reason that it is expected to be multiplied by `dR`. Therefore dD = (d Rfunc(r)/dr)/r` rater than just `d Rfunc(r)/dr`
* you are right invR2 si 1/R0^2 not 1/r^2, I overlooked that while replying you. But still the code is fine (I think):

```
E(r) = 1/( r^6 + Rfunc(r) )
F(r) =  dR *   -E(r)^2 * ( 6*r^4 + (dRfunc(r)/dr)/r )
```

* ad sign - I always make mistakes in this. Never shure if we store `F=-dE/dr` or `F=dE/dr`. But notice that the test here works (meaning the sign convecction is the same as for un-damped L-J and vdW )
https://github.com/Probe-Particle/ppafm/issues/48#issuecomment-1429919012

please try tu run the test here if you don't believe me:
https://github.com/Probe-Particle/ppafm/blob/prokop-damp-func/dev/test_vdwDaping.py

Reply:
Ah, I didn't see that `dD` is missing a factor of `r`. Then it makes sense. I'm still a bit confused by the sign convention, but it looks like it comes out correct in the end, so I guess it's fine.